[Bug]: Submitting an incorrect password throws an unhandled rejection in the show private key flow

### Describe the bug


https://github.com/MetaMask/metamask-extension/assets/53189696/679a3fb6-af5a-43d3-a7af-d6f279b43cac



### Steps to reproduce

1. log in to the extension
2. click the account menu icon
3. click the ellipsis icon to the right of the first account in the account list
4. select the Account Details menu item
5. click the Show Private Key button
6. enter an incorrect password, click Confirm
7. wait a few seconds


### Error messages or log output

Error
```shell
sentry-install.js:1 Error in verifying password.
e.<computed> @ sentry-install.js:1
(anonymous) @ sentry-install.js:1
error @ runtime-lavamoat.js:7114
(anonymous) @ ui-12.js:1
handleResponse @ ui-12.js:1
(anonymous) @ common-3.js:7
s.emit @ common-3.js:7
w @ common-5.js:3
_ @ common-5.js:3
y.push @ common-5.js:3
_write @ common-4.js:26
y @ common-5.js:3
(anonymous) @ common-5.js:3
g.write @ common-5.js:3
m @ common-5.js:3
(anonymous) @ common-3.js:7
s.emit @ common-3.js:7
w @ common-5.js:3
_ @ common-5.js:3
y.push @ common-5.js:3
_onMessage @ common-3.js:16
(anonymous) @ common-3.js:16
```

Unhandled exception
```shell
sentry-install.js:1 SES_UNHANDLED_REJECTION: (Error#6)
e.<computed> @ sentry-install.js:1
(anonymous) @ sentry-install.js:1
error @ runtime-lavamoat.js:7114
handleRejection @ runtime-lavamoat.js:7415
finalizeDroppedPromise @ runtime-lavamoat.js:7243
```

```shell
sentry-install.js:1 Error#6: Incorrect password
e.<computed> @ sentry-install.js:1
(anonymous) @ sentry-install.js:1
logError @ runtime-lavamoat.js:7075
logSubErrors @ runtime-lavamoat.js:7018
error @ runtime-lavamoat.js:7116
handleRejection @ runtime-lavamoat.js:7415
finalizeDroppedPromise @ runtime-lavamoat.js:7243
sentry-install.js:1   at c (chrome-extension://cmbgfohndfdmokhhionelhibkgpkmllf/common-0.js:1:129596)
  at async Object.o [as decrypt] (chrome-extension://cmbgfohndfdmokhhionelhibkgpkmllf/common-0.js:1:129365)
  at async T.verifyPassword (chrome-extension://cmbgfohndfdmokhhionelhibkgpkmllf/background-0.js:13:101987)
  at async Ke.verifyPassword (chrome-extension://cmbgfohndfdmokhhionelhibkgpkmllf/background-4.js:1:209385)
  at async s.<anonymous> (chrome-extension://cmbgfohndfdmokhhionelhibkgpkmllf/background-4.js:1:56518)
```


### Version

11.1.0

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

--- a/ui/components/multichain/account-details/account-details-authenticate.js
+++ b/ui/components/multichain/account-details/account-details-authenticate.js
@@ -7,6 +7,8 @@ import {
   Severity,
   TextVariant,
 } from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { exportAccount, hideWarning } from '../../../store/actions';
 import {
   BannerAlert,
   Box,
@@ -14,9 +16,6 @@ import {
   ButtonSecondary,
   FormTextField,
 } from '../../component-library';
-
-import { useI18nContext } from '../../../hooks/useI18nContext';
-import { exportAccount, hideWarning } from '../../../store/actions';
 
 export const AccountDetailsAuthenticate = ({
   address,
@@ -35,10 +34,14 @@ export const AccountDetailsAuthenticate = ({
   const onSubmit = useCallback(() => {
     dispatch(
       exportAccount(password, address, setPrivateKey, setShowHoldToReveal),
-    ).then((res) => {
-      dispatch(hideWarning());
-      return res;
-    });
+    )
+      .then((res) => {
+        dispatch(hideWarning());
+        return res;
+      })
+      .catch(() => {
+        // No need to do anything more with the caught error here, we already logged the error
+      });
   }, [dispatch, password, address, setPrivateKey, setShowHoldToReveal]);
 
   const handleKeyPress = useCallback(
@@ -57,7 +60,7 @@ export const AccountDetailsAuthenticate = ({
         id="account-details-authenticate"
         label={t('enterYourPassword')}
         placeholder={t('password')}
-        error={warning}
+        error={Boolean(warning)}
         helpText={warning}
         onChange={(e) => setPassword(e.target.value)}
         value={password}

--- a/ui/pages/keychains/reveal-seed.js
+++ b/ui/pages/keychains/reveal-seed.js
@@ -133,7 +133,7 @@ export default function RevealSeedPage() {
           size={TEXT_FIELD_SIZES.LG}
           value={password}
           onChange={(event) => setPassword(event.target.value)}
-          error={error}
+          error={Boolean(error)}
           width={BlockSize.Full}
         />
         {error && (

